backend/src/jv2backend/requestData.py

```
@@ -0,0 +1,132 @@
+        self._is_http: bool = False
+        self._error: str = None
+
+        # Root url is always required
+        if "rootUrl" not in requestData:
+            self._error = "No root URL provided in request."
+            return
+        self._root_url = requestData["rootUrl"]
+
+        # Determine whether this is a network or file url
+        self._is_http = (self._root_url.lower().startswith("http://") or
+                         self._root_url.lower().startswith("https://"))
+
+        # Assess the full path / library key - may have optional "directory"
+        if "directory" in requestData:
+            self._directory = requestData["directory"]
+        self._full_url = (self._root_url if self._directory is None else
+                          url_join(self._root_url, self._directory))
+
+        # Try to find the library corresponding to the library key
+        self._journal_collection = (library[self._full_url] if
+                                    self._full_url in library else None)
+        if require_in_library and self._journal_collection is None:
+            self._error = f"No collection {self._full_url} in library."
+            return
+
+        # Was a data directory provided?
+        self._data_directory = (requestData["dataDirectory"]
+                                if "dataDirectory" in requestData else None)
+
+        # Was a data directory required?
+        if require_data_directory and self._data_directory is None:
+            self._error = f"Data directory required for URL \
+                          {self._full_url}."
+            return
+
+        # Was an optional index file provided?
+        self._index_file = (requestData["indexFile"] if "indexFile" in
+                            requestData else None)
+
+        # Was it required?
+        if require_index_for_http and self._index_file is None:
+            self._error = f"Index file required for URL {self._full_url}."
+            return
```

To add on to what I said before, here you can raise an exception instead of returning, I think that it would be a clearer way of showing that the object initialisation failed.

E.g:

```suggestion
            raise InvalidRequest(f"Index file required for URL {self._full_url}.", status_code=400)
```